bread/layout/components/search.py

```diff
@@ -1,69 +1,112 @@
+import typing
+
 import htmlgenerator as hg
 from django.utils.translation import gettext_lazy as _
 
 from .icon import Icon
 from .loading import Loading
 
 
+class SearchBackendConfig(typing.NamedTuple):
+    url: typing.Any
+    result_selector: str = None
+    result_label_selector: str = None
+    result_value_selector: str = None
+    query_parameter: str = "q"
```

Comment:
I would be interested what your reasoning behind extracting a config-object is here. I definitely see the case that we need the search-backend to more configurable.

(Background of why I am asking a bit more is: I am always a bit suspicious of factoring parameter objects out: In the worst case we just changed ```func(param1, param2, param3)``` into ```func(FuncParam(param1, param2, param3))``` which is actually worse because now we first need to understand what a ```FuncParam``` object is. If is just about grouping parameters, then using a prefix is sometimes more readable. And it helps that Python has rather flexible parameter handling.)

Reply:
The motivation is exactly the same as for extracting functions.
Also with extracting functions your argument is valid: I agree that changing
`stmt1; 
stmt2; 
stmt3` 
to 
`foo();
def foo():
    stmt1;
    stmt2;
    stmt3;
`
is not a benefit. But if we have something like `stmt1;...stmt14;`, and `stmt4;...stmt8;` are more related to each other than to the rest, it can be an improvement to extract `stmt4;...stmt8;` into a separate method. In my opinion this applies to any kind of data. As soon as tuples, argument lists, etc. reach a certain length, it can be helpful to add more structure.

Reply:
I agree that it can be helpful, but as long as the sole purpose is grouping parameters I will stay skeptic... Better naming could be an alternative as well (I think that naming does not get attention it deserve sometimes ;)
Of course when passing config-objects around or having config-generators/shortcuts or similar "consumers" of the additional API then I see a more favorable case for factoring parameters out. And actually I see that to be a good use case for the search backends: Basically that a view could e.g. provide its own SearchBackendConfig.

But anyway, just bringing in my opinion here, no reason to stop merging.